src/Blocks/sources.jl

```diff
@@ -145,26 +283,81 @@ Generate exponentially damped sine signal.
 - `frequency`: [Hz] Frequency of sine wave
 - `amplitude`: Amplitude of sine wave
 - `damping`: [1/s] Damping coefficient of sine wave
-- `phase`: [rad] Phase of sine wave 
+- `phase`: [rad] Phase of sine wave
 - `offset`: Offset of output signal
 - `start_time`: [s] Output `y = offset` for `t < start_time`
+- `smooth`:  If `true`, returns a smooth wave. Defaults to `false`
+             It uses a smoothing factor of `δ=1e-5`
 
 # Connectors:
 - `output`
 """
-function ExpSine(;name, 
-    frequency, 
+function ExpSine(; name,
+    frequency,
     amplitude=1,
     damping=0.1,
     phase=0,
     offset=0,
-    start_time=0)
+    start_time=0,
+    smooth=false)
 
     @named output = RealOutput()
     pars = @parameters offset=offset start_time=start_time amplitude=amplitude frequency=frequency phase=phase damping=damping
+
+    equation = if smooth == false
+        offset + ifelse(t < start_time, 0, amplitude * exp(-damping * (t - start_time)) * sin(2*pi*frequency*(t - start_time) + phase))
+    else
+        δ = 1e-5
+        smooth_damped_sin(t, δ, frequency, amplitude, damping, phase, offset, start_time)
+    end
+
+    eqs = [
+        output.u ~ equation
+    ]
+
+    compose(ODESystem(eqs, t, [], pars; name=name), [output])
+end
+
+"""
+Generate smooth triangular signal for frequencies less than or equal to 25 Hz
+
+# Parameters:
+- `frequency`: [Hz] Frequency of square wave
+- `amplitude`: Amplitude of square wave
+- `offset`: Offset of output signal.
+- `start_time`: [s] Output `y = offset` for `t < start_time`
+- `smooth`:  If `true`, returns a smooth wave. Defaults to `false`
+             It uses a smoothing factor of `δ=1e-5`
+
+# Connectors:
+- `output`
+"""
+function Triangular(; name, amplitude=1.0, frequency=1.0,
+    offset=0.0, start_time=0.0, smooth=false)
+
+    if smooth
+        frequency > 25 && @warn "`frequency > 25` can lead to non-triangular wave pattern"
```

Comment:
What's up with this?

Reply:
Perhaps the smoothing constants need to be frequency dependent?

Reply:
![f=25, 26, 30](https://user-images.githubusercontent.com/32921645/171209570-8d5c8aa4-ddf6-4208-a544-70dbf53b8a7b.png)

For f>25, the waveform is totally erratic

>Perhaps the smoothing constants need to be frequency dependent?

I tried these:
```
amplitude * (1-2acos((1 - f* δ)sin(2π*(x - start_time)*f))/π)
amplitude * (1-2acos((1 - δ/f)sin(2π*(x - start_time)*f))/π)
amplitude * (1-2acos((1 - δ)sin(2π*(x - start_time)*f*δ))/π)
amplitude * (1-2acos((1 - δ)sin(2π*(x - start_time)*f/δ))/π)
amplitude * (1-2acos(sin(2π*(x - start_time)*f)/π)
```

None of these relations gave a better result.
I think it comes down to trigonometric functions used. Is there any source I can look at for this?

Reply:
Can you try evaluating the functions explicitly? Maybe it is just a solver issue.

Reply:
This is still open which is why I didn't merge. What's the consensus here? I didn't look into it.

Reply:
Aha, the real issue is the stepsize. And this is present in both `smooth=true/false` wave.

for `t = 0:0.001:1`

![tri-smooth-f=26](https://user-images.githubusercontent.com/32921645/171844094-544507ec-3c21-48a6-a535-f045be2fb247.png)


Should I mention a note on this in docstring?

(Will remove the `@warn`)

Reply:
Adaptivity should handle it?

Reply:
Somehow it didn't. For whenever `offset!=0` similar issue crops up. `saveat=0.01` saves. ([Here](https://github.com/SciML/ModelingToolkitStandardLibrary.jl/blob/55668b14c365562b592be1fb8d805e4edc860f16/test/Blocks/sources.jl#L270))